src/Metal/MTLArgumentEncoder.cs

```diff
@@ -1,11 +1,54 @@
 using System;
+using System.Runtime.InteropServices;
+using Foundation;
+using ObjCRuntime;
 
 namespace Metal {
+
+#if !XAMCORE_4_0
+	public partial interface IMTLArgumentEncoder {
+		[Obsolete ("Use 'SetManagedBuffers' instead.")]
+		void SetBuffers (IMTLBuffer [] buffers, IntPtr offsets, NSRange range);
+	}
+
+	internal partial class MTLArgumentEncoderWrapper {
+
+		[Obsolete ("Use 'SetManagedBuffers' instead.")]
+		public void SetBuffers (IMTLBuffer [] buffers, IntPtr offsets, NSRange range) 
+			=> throw new NotSupportedException ();
+	}
+#endif
+
 	public static partial class MTLArgumentEncoder_Extensions {
+#if !XAMCORE_4_0
+		[Obsolete ("Use 'SetManagedBuffers' instead.")]
 		public unsafe static void SetBuffers (this IMTLArgumentEncoder This, IMTLBuffer [] buffers, nint [] offsets, Foundation.NSRange range)
+			=> throw new NotSupportedException ();
+#endif
+
+#if XAMCORE_4_0
+		public static void SetBuffers (this IMTLArgumentEncoder table, IMTLBuffer[] buffers, nuint[] offsets, NSRange range)
+#else
+		public static void SetManagedBuffers (this IMTLArgumentEncoder table, IMTLBuffer[] buffers, nuint[] offsets, NSRange range)
```

Comment:
Why can't you just use `SetBuffers` always?

Reply:
With ArgumentEncode I think we can not with the other.

Reply:
I don't see why not, because there will be only 1 method with this signature. Does it fail to compile if you try?